crates/s3s/src/http/body.rs

```diff
@@ -31,7 +35,11 @@ pin_project_lite::pin_project! {
         },
         Hyper {
             #[pin]
-            inner: hyper::Body,
+            inner: hyper::body::Incoming,
+        },
+        BoxBody {
+            #[pin]
+            inner: BoxBody,
         },
```

Comment:
should we merge the Hyper into the BoxBody variant?

Reply:
Merging `Hyper` adds an extra boxing for each request but reduces branching for each frame. It's OK to merge since there are already too many allocations for each request.

Reply:
will leave it as is for now. I think this can wait for the Body refactor.